src/compiler/parsing/watf/grammar.js

```diff
@@ -193,6 +194,48 @@ export function parse(tokensList: Array<Object>, source: string): Program {
       return t.memory(limits, id);
     }
 
+    /**
+     * Parses a data section
+     * https://webassembly.github.io/spec/core/text/modules.html#data-segments
+     *
+     * WAST:
+     *
+     * data:  ( data <index>? <offset> <string> )
+     */
+    function parseData(): Data {
+      eatTokenOfType(tokens.openParen);
+
+      let offset: Instruction;
+      if (token.type === tokens.valtype) {
```

Comment:
the logic I've added to parse a `const` or `get_global` feels like it must be repeating some code elsewhere - but I just can't find the right code to re-use!

Reply:
Yeah that's right, I would prefer to implement the parser (+ tests) before trying to re-use code.

Reply:
cool, makes sense.